tree/criterion.hpp

```diff
@@ -0,0 +1,30 @@
+#ifndef MACHINELEARNING_TREE_CRITERION_HPP_
+#define MACHINELEARNING_TREE_CRITERION_HPP_
+
+namespace tree {
+
+/**
+ * \brief Base class that defines the many criterious of splitting.
+ *
+ * This interface provides some methods that are used to compute the impurity of a given Node object.
+ * Some impurity measures are Gini, Entrophy and Missclassification Error Rate for classification problens and MSE for regression
```

Comment:
Line length must be at most 80 chars. Please check https://google.github.io/styleguide/cppguide.html#Line_Length.